### PR TITLE
Add Buildifier support for Bazel CI

### PR DESCRIPTION
**Pull request by fweikert**
This PR allows users to run Buildifier as a linter for all builds of selected Buildkite pipelines. Buildifier can be enabled and configured via the respective pipeline configuration yaml file.

See buildkite/README.md for more details.

### CODE REVIEW

**Comment by fweikert** [comment]
cc @alexeagle

**Comment by fweikert** [comment]
TODO: format the changes

**Comment by fweikert** [comment]
Fixes #338

File: buildkite/bazelci.py [hunk 1]
```diff
@@ -309,6 +311,16 @@
...
 
+LATEST_VERSION_PATTERN = re.compile(r"latest(-(?P<offset>\d+))?$")
+
+BUILDIFIER_RELEASE_PAGE = "https://api.github.com/repos/bazelbuild/buildtools/releases"
+
+BUILDIFIER_DEFAULT_VERSION = "latest"
+
+BUILDIFIER_DEFAULT_INPUT_FILES = ["BUILD", "*.bzl"]
+
+BUILDIFIER_DEFAULT_DOCKER_IMAGE = "gcr.io/bazel-untrusted/ubuntu1804:nojava"
```
Comment (philwo) [inline buildkite/bazelci.py]: Can we just rename this to BUILDIFIER_DOCKER_IMAGE? It's not configurable, so it's not really a "default".

File: buildkite/bazelci.py [hunk 2]
```diff
@@ -309,6 +311,16 @@
     },
 }
 
+LATEST_VERSION_PATTERN = re.compile(r"latest(-(?P<offset>\d+))?$")
+
+BUILDIFIER_RELEASE_PAGE = "https://api.github.com/repos/bazelbuild/buildtools/releases"
+
+BUILDIFIER_DEFAULT_VERSION = "latest"
```
[the remark this answers is not in the document]
Comment (fweikert) [reply buildkite/bazelci.py]: I talked to Vlad, and he confirmed that there is no correlation with Bazel versions, i.e. using the latest should be fine. However, that means we need to update the docker images frequently.

[the remark this answers is not in the document]
Comment (philwo) [reply buildkite/bazelci.py]: > that means we need to update the docker images frequently.

That's fine - I think we can even build them on Buildkite automatically every day or so with just a small amount of work! :)

File: buildkite/bazelci.py [hunk 3]
```diff
@@ -309,6 +311,16 @@
...
+
+BUILDIFIER_DEFAULT_INPUT_FILES = ["BUILD", "*.bzl"]
```
[the remark this answers is not in the document]
Comment (fweikert) [reply buildkite/bazelci.py]: Done.

**Pull request merged**